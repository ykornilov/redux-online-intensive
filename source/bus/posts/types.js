// Sync
export const FILL_POSTS = 'FILL_POSTS';
export const ADD_NEW_POST = 'ADD_NEW_POST';

// Async
export const FETCH_POSTS_ASYNC = 'FETCH_POSTS_ASYNC';
export const CREATE_POST_ASYNC = 'CREATE_POST_ASYNC';
