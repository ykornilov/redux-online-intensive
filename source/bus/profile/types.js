export const types = {
    // Sync
    FILL_PROFILE:  'FILL_PROFILE',
    UPDATE_AVATAR: 'UPDATE_AVATAR',
    CLEAR_PROFILE: 'CLEAR_PROFILE',

    // Async
    UPDATE_NAME_ASYNC:     'UPDATE_NAME_ASYNC',
    UPDATE_AVATAR_ASYNC:   'UPDATE_AVATAR_ASYNC',
    UPDATE_PASSWORD_ASYNC: 'UPDATE_PASSWORD_ASYNC',
};
