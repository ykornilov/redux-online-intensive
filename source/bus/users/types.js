export const types = {
    // Sync
    FILL_USERS:  'FILL_USERS',
    CLEAR_USERS: 'CLEAR_USERS',

    // ASYNC
    FETCH_USERS_ASYNC: 'FETCH_USERS_ASYNC',
};
