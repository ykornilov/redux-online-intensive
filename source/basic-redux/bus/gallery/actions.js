import { SHOW_PREV_PHOTO, SHOW_NEXT_PHOTO, SHOW_SELECTED_PHOTO } from './types';

export const showPrevPhoto = () => {
    return {
        type: SHOW_PREV_PHOTO,
    };
};

export const showNextPhoto = () => {
    return {
        type: SHOW_NEXT_PHOTO,
    };
};

export const showSelectedPhoto = (photoIndex) => {
    return {
        type:    SHOW_SELECTED_PHOTO,
        payload: photoIndex,
    };
};
