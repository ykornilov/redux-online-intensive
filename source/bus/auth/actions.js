// Types
import { types } from './types';

export const authActions = {
    // Sync
    authenticate: () => {
        return {
            type: types.AUTHENTICATE,
        };
    },

    // Async
    signupAsync: (userData) => {
        return {
            type:    types.SIGNUP_ASYNC,
            payload: userData,
        };
    },
    loginAsync: (loginData) => {
        return {
            type:    types.LOGIN_ASYNC,
            payload: loginData,
        };
    },
};
