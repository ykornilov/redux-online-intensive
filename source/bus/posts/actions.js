// Types
import { FETCH_POSTS_ASYNC, FILL_POSTS } from './types';

// Instuments
import { api } from '../../REST';

export const fillPosts = (posts) => {
    return {
        type:    FILL_POSTS,
        payload: posts,
    };
};

export const fetchPostsAsync = () => async (dispatch) => {
    dispatch({
        type: FETCH_POSTS_ASYNC,
    });

    const response = await api.posts.fetch();
    const result = await response.json();

    dispatch(fillPosts(result.data));
};
