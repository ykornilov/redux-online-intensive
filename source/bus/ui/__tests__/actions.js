// Types
import { types } from '../types';

// Actions
import { uiActions } from '../actions';

describe('UI actions:', () => {
    test('startFetching', () => {
        expect(uiActions.startFetching()).toEqual({
            type: types.START_FETCHING,
        });
    });

    test('stopFetching', () => {
        expect(uiActions.stopFetching()).toEqual({
            type: types.STOP_FETCHING,
        });
    });

    test('setOnlineState', () => {
        expect(uiActions.setOnlineState()).toEqual({
            type: types.SET_ONLINE_STATE,
        });
    });

    test('setOfflineState', () => {
        expect(uiActions.setOfflineState()).toEqual({
            type: types.SET_OFFLINE_STATE,
        });
    });

    test('emitError', () => {
        expect(uiActions.emitError(new Error())).toEqual({
            type:    types.EMIT_ERROR,
            payload: new Error(),
            error:   true,
            meta:    null,
        });
    });
});
