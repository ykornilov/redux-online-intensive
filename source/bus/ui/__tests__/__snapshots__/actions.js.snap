// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI actions: emitError 1`] = `
Object {
  "error": true,
  "meta": null,
  "payload": [Error],
  "type": "EMIT_ERROR",
}
`;

exports[`UI actions: emitError with meta-data 1`] = `
Object {
  "error": true,
  "meta": "worker",
  "payload": [Error],
  "type": "EMIT_ERROR",
}
`;

exports[`UI actions: setOfflineState 1`] = `
Object {
  "type": "STATE_OFFLINE_STATE",
}
`;

exports[`UI actions: setOnlineState 1`] = `
Object {
  "type": "STATE_ONLINE_STATE",
}
`;

exports[`UI actions: startFetching 1`] = `
Object {
  "type": "START_FETCHING",
}
`;

exports[`UI actions: stopFetching 1`] = `
Object {
  "type": "STOP_FETCHING",
}
`;
